Source/cmDebuggerNamedPipeWin32.cxx

```diff
@@ -0,0 +1,121 @@
+/* Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+   file Copyright.txt or https://cmake.org/licensing for details.  */
+
+#include "cmsys/Encoding.hxx"
+
+#include "cmDebuggerNamedPipe.h"
+#include "cmSystemTools.h"
+
+namespace cmDebugger {
+
+cmDebuggerNamedPipe::cmDebuggerNamedPipe(std::string const& name)
```

Comment:
Is this constructor used?

Reply:
Yes, used in cmDebuggerAdapterFactory.cxx.

![image](https://user-images.githubusercontent.com/105310954/224843561-03c3a417-40e0-4ab0-b251-11418f3563bb.png)

Reply:
@yuehuang010 I've refactored the code a bit, got rid of the cmDebuggerNamedPipe.h and hide the platform-specific details in the implementation files as much as possible, please take another look, thanks.